File: backend/app/models/template.py
"""
编号模板模型 - 零件号自动生成规则配置

编号模板定义零件号的生成格式，由编号引擎（number_engine）使用。
示例格式：PREFIX-TYPE_CODE-SUBSYSTEM-001
"""
import uuid
from datetime import datetime
from sqlalchemy import Column, String, Integer, DateTime, ForeignKey
from app.utils.types import PortableUUID as UUID, PortableJSON as JSONB

from app.database import Base


class PartNumberTemplate(Base):
    """零件号模板，配置前缀、分隔符、序号位数等编号规则"""

    __tablename__ = "part_number_templates"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    # 模板名称，如"2026赛季零件编号"
    name = Column(String(100), nullable=False)
    # 编号前缀，如"FSAE"
    prefix = Column(String(20))
    # 各段之间的分隔符，默认 "-"
    separator = Column(String(5), default="-")
    # 序号的零填充位数，如 3 表示 001、002...
    digit_count = Column(Integer, default=3)
    # 子系统代码映射，如 {"动力": "ENG", "底盘": "CHS"}
    subsystem_codes = Column(JSONB, default=dict)
    # 类型代码映射，如 {"part": "PRT", "assembly": "ASM"}
    type_codes = Column(JSONB, default=dict)
    created_by = Column(UUID(as_uuid=True), ForeignKey("users.id"))
    created_at = Column(DateTime, default=datetime.utcnow)

File: catia_client_cs/CatiaClient/Models/Template.cs
namespace CatiaClient.Models;

/// <summary>
/// 零件编号模板，定义编号生成规则。
/// 例如模板 "SUS" 可生成编号 "SUS-A-001"，其中 A 为子系统代码，001 为序号。
/// </summary>
public class Template
{
    /// <summary>模板唯一标识符</summary>
    public Guid Id { get; set; }

    /// <summary>模板名称（如悬架模板、转向模板）</summary>
    public string Name { get; set; } = "";

    /// <summary>编号前缀（如 SUS、STR）</summary>
    public string? Prefix { get; set; }

    /// <summary>编号各段之间的分隔符，默认为 "-"</summary>
    public string Separator { get; set; } = "-";

    /// <summary>序号位数，默认为 3（生成 001、002 等）</summary>
    public int DigitCount { get; set; } = 3;

    /// <summary>子系统名称到代码的映射（如 "悬架" -> "A"）</summary>
    public Dictionary<string, string> SubsystemCodes { get; set; } = new();

    /// <summary>零件类型到代码的映射（如 "零件" -> "P"、"总成" -> "A"）</summary>
    public Dictionary<string, string> TypeCodes { get; set; } = new();
}
